ksqldb-functional-tests/src/test/resources/query-validation-tests/avro.json

```diff
@@ -1183,6 +1183,90 @@
+        "CREATE STREAM MODEL_TXS_EVENT WITH  (KAFKA_TOPIC='model-transaction', KEY_FORMAT='KAFKA', VALUE_FORMAT='AVRO', VALUE_SCHEMA_ID=102) AS SELECT `id`, as_value(txs.`id`) AS `eventID`, STRUCT(`num_shares` := CAST(EXTRACTJSONFIELD(txs.`transaction`, '$.num_shares') AS INT), `amount` := CAST(EXTRACTJSONFIELD(txs.`transaction`, '$.amount') AS INT), `customer` := STRUCT(`first_name`:= EXTRACTJSONFIELD(txs.`transaction`, '$.customer.first_name'), `last_name`:= EXTRACTJSONFIELD(txs.`transaction`, '$.customer.last_name'), `email`:= EXTRACTJSONFIELD(txs.`transaction`, '$.customer.email'))) AS `transaction` FROM TXS_REKEY txs;"
+      ],
+      "topics": [
+        {
+          "name": "model-transaction",
+          "valueFormat": "AVRO",
+          "valueSchemaId": 102,
+          "valueSchema": {"fields":[{"name":"eventID","type":"string"},{"name":"transaction","type":{"fields":[{"name":"num_shares","type":"int","default": 0},{"name":"amount","type":"int","default": 0},{"name":"customer","type":{"fields":[{"name":"first_name","type":"string"},{"name":"last_name","type":"string"},{"name":"email","type":"string"}],"name":"customer","type":"record"}}],"name":"transaction","type":"record"}}],"name":"Transaction","namespace":"com.acme.namespace","type":"record"}
+        }
+      ],
+      "inputs": [
+        {"topic": "transaction", "key": null, "value": {"id": "1", "transaction": {"txn_ts": "2020-11-18 02:31:43", "customer": {"first_name": "Jill", "last_name": "Smith", "id": 1234567, "email": "jsmith@foo.com" }, "company": {"name": "ACME Corp", "ticker": "ACMC", "id": "ACME837275222752952", "address": "Anytown USA, 333333"}}}},
+        {"topic": "transaction", "key": null, "value": {"id": "2", "transaction": {"txn_ts": "2020-11-18 02:35:43", "customer": { "first_name": "Art", "last_name": "Vandeley", "id": 8976612, "email": "avendleay@foo.com" }, "company": { "name": "Imports Corp", "ticker": "IMPC", "id": "IMPC88875222752952", "address": "Anytown USA, 333333"}}}},
+        {"topic": "transaction", "key": null, "value": {"id": "3", "transaction": {"txn_ts": "2020-11-18 02:36:43", "customer": { "first_name": "John", "last_name": "England", "id": 456321, "email": "je@foo.com" }, "company": { "name": "Hechinger", "ticker": "HECH", "id": "HECH8333785222752952", "address": "Anytown USA, 333333"}}}},
+        {"topic": "transaction", "key": null, "value": {"id": "4", "transaction": {"txn_ts": "2020-11-18 02:37:43", "customer": { "first_name": "Fred", "last_name": "Pym", "id": 333567, "email": "fjone@foo.com" }, "company": { "name": "PymTech", "ticker": "PYMT", "id": "PYME837275222714197419202020", "address": "Anytown USA, 333333"}}}}
+      ],
+      "outputs": [
+        {"topic": "model-transaction", "key": "1", "value": {"eventID": "1", "transaction": {"amount": 0, "num_shares": 0, "customer": {"last_name": "Smith", "first_name": "Jill", "email": "jsmith@foo.com"}}}},
```

Comment:
Curious if this amount:0 is from `CAST(EXTRACTJSONFIELD(txs.`transaction`, '$.amount') AS INT)`? Since physical schema and logical schema both have `amount` field, it's not from default value?

Reply:
No, I tried with different default values and finally set it as 0 for the test.

Reply:
Is it because `CAST(EXTRACTJSONFIELD(txs.transaction, '$.amount') AS INT` is `null` so the default value is picked? I'm trying to understand why it used default value since the `amount` field is in both schemas and `addDefaultValueOrThrowException` won't be hit

Reply:
Added a test to cover addDefaultValueOrThrowException method